packages/tooltips/src/elements/Tooltip.tsx

```diff
@@ -40,43 +43,54 @@ const TooltipWrapper = styled.div`
   z-index: ${props => props.zIndex};
 `;
 
-const SIZE = {
-  SMALL: 'small',
-  MEDIUM: 'medium',
-  LARGE: 'large',
-  EXTRA_LARGE: 'extra-large'
-};
-
-const TYPE = {
-  LIGHT: 'light',
-  DARK: 'dark'
-};
-
-/**
- * Due to Popper.JS needing a reference to a component we provide a simple wrapper
- * to ensure the correct reference is provided.
- */
-const TriggerWrapper = styled.div`
-  display: inline-block;
-`;
+type TOOLTIP_TYPE = 'light' | 'dark';
 
-const Tooltip = ({
-  appendToBody,
-  arrow,
-  children,
-  delayMilliseconds,
-  eventsEnabled,
+export interface ITooltipProps
+  extends Partial<ThemeProps<DefaultTheme>>,
+    React.HTMLAttributes<HTMLDivElement> {
+  /** Appends the tooltip to the body element */
+  appendToNode?: Element;
+  showArrow?: boolean;
+  /** Milliseconds of delay before open/close of tooltip is initiated */
+  delayMilliseconds?: number;
+  /** Whether Popper.js should update based on DOM resize events */
+  eventsEnabled?: boolean;
+  id?: string;
+  content: React.ReactNode;
+  /**
+   * These placements differ from the default naming of Popper.JS placements to help
+   * assist with RTL layouts.
+   **/
+  placement?: GARDEN_PLACEMENT;
+  /** Passes options to [Popper.JS Instance](https://github.com/FezVrasta/popper.js/blob/master/docs/_includes/popper-documentation.md#new-popperreference-popper-options) */
+  popperModifiers?: any;
+  size?: TOOLTIP_SIZE;
+  type?: TOOLTIP_TYPE;
+  /** The z-index of the popper.js placement container */
+  zIndex?: number | string;
+  initialIsVisible?: boolean;
+  children: React.ReactElement;
+  refKey?: string;
+}
+
+const Tooltip: React.FC<ITooltipProps> = ({
   id,
-  trigger,
+  delayMilliseconds,
+  initialIsVisible,
+  content,
+  refKey,
   placement,
+  eventsEnabled,
   popperModifiers,
+  children,
+  showArrow,
   size,
   type,
+  appendToNode,
   zIndex,
-  initialIsVisible,
   ...otherProps
 }) => {
-  const scheduleUpdateRef = useRef();
+  const scheduleUpdateRef = useRef<any>();
```

Comment:
Do you think there is any benefit of typing this as `() => void` rather than `any`? 

TypeScript compiler passes with:

```ts
const scheduleUpdateRef = useRef<() => void>();
```

Reply:
This works! When I originally moved this into TS `scheduleUpdate` was typed as an `any` within `react-popper`. This now matches their value.